src/app/public/modules/overlay/overlay.service.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ComponentRef,
+  Injectable,
+  Type
+} from '@angular/core';
+
+import {
+  SkyDynamicComponentService
+} from '../dynamic-component';
+
+import {
+  SkyOverlayConfig
+} from './overlay-config';
+
+import {
+  SkyOverlayDomAdapterService
+} from './overlay-dom-adapter.service';
+
+import {
+  SkyOverlayHostComponent
+} from './overlay-host.component';
+
+import {
+  SkyOverlayInstance
+} from './overlay-instance';
+
+/**
+ * This service is used to create new overlays.
+ */
+@Injectable()
+export class SkyOverlayService {
+
+  private host: ComponentRef<SkyOverlayHostComponent>;
+
+  private instances: SkyOverlayInstance<any>[] = [];
+
+  constructor(
+    private dynamicComponentService: SkyDynamicComponentService,
+    private adapter: SkyOverlayDomAdapterService
+  ) {
+    this.createHostComponent();
+  }
+
+  /**
+   * Creates a new overlay and appends an instance of the provided component.
+   * @param component The component to append to the overlay.
+   * @param config The configuration for the overlay.
+   */
+  public create<T>(
+    component: Type<T>,
+    config: SkyOverlayConfig = {
```

Comment:
Added the default values for `SkyOverlayConfig` to the method parameter so that it gets picked up by the IDE and our documentation tools.